includes/abstracts/abstract-wc-object-query.php

```diff
@@ -0,0 +1,83 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Abstract WC Object Query Class
+ *
+ * Extended by classes to provide a query abstraction layer for safe object searching.
+ *
+ * @version  3.1.0
+ * @package  WooCommerce/Abstracts
+ * @category Abstract Class
+ * @author   Automattic
+ */
+abstract class WC_Object_Query {
+
+	/**
+	 * Stores query data.
+	 * @var array
+	 */
+	public $query_vars = array();
```

Comment:
Yeah. I think I was trying to make this too similar to WP_Query. I'll try and steer away from that from now on. :)